src/Services/ShippingService.php

```diff
@@ -32,4 +36,72 @@ public function setShippingProfileId(int $shippingProfileId)
 	{
 		$this->checkout->setShippingProfileId($shippingProfileId);
 	}
+    
+    /**
+     * Return an array of shipping profile ids as keys and max delivery days as values
+     * @return array
+     */
+	public function getMaxDeliveryDays()
+    {
+        $maxDeliveryDays = [];
+        
+        /** @var CheckoutService $checkoutService */
+        $checkoutService = pluginApp(CheckoutService::class);
+        $shippingProfileList = $checkoutService->getShippingProfileList();
+        
+        if(count($shippingProfileList))
+        {
+            $currentShippingCountryId = $checkoutService->getShippingCountryId();
+            /** @var CountryRepositoryContract $countryRepo */
+            $countryRepo = pluginApp(CountryRepositoryContract::class);
+            /** @var Country $country */
+            $country = $countryRepo->getCountryById($currentShippingCountryId);
+    
+            /** @var ParcelServicePresetRepositoryContract $parcelServicepresetRepo */
+            $parcelServicepresetRepo = pluginApp(ParcelServicePresetRepositoryContract::class);
+    
+            foreach($shippingProfileList as $shippingProfile)
+            {
+                $parcelServicePreset = $parcelServicepresetRepo->getPresetById($shippingProfile['parcelServicePresetId']);
+        
+                if($parcelServicePreset instanceof ParcelServicePreset)
+                {
+                    $generalSettings = $parcelServicePreset->parcelServiceRegionConstraint->where('shippingRegionId', $country->shippingDestinationId)->first()->constraint->first()->generalSettings;
```

Comment:
Hier könnte man sicherheitshalber noch den Fall abfangen, wenn `$parcelServicePreset->parcelServiceRegionConstraint->where('shippingRegionId', $country->shippingDestinationId)` keine Einträge zurück liefert.

In dem Fall am besten einen Log-Eintrag schreiben!